src/VisualStudio/IntegrationTest/TestSetup/AsyncCompletionTracker.cs

```diff
@@ -48,7 +50,21 @@ private void HandleAsyncCompletionTriggered(object sender, CompletionTriggeredEv
 
             e.CompletionSession.Dismissed += ReleaseToken;
             e.CompletionSession.ItemCommitted += ReleaseToken;
-            e.CompletionSession.ItemsUpdated += ReleaseToken;
+
+            _ = Task.Run(async () =>
+                {
+                    // AsyncCompletion might fires multiple ItemsUpdated events perf keystroke typed, which means
+                    // we could see the first ItemsUpdated event even though items don't change (but computation finished).
```

Comment:
wonder if this is the same flakiness issue thats impacting the completion speedometer as well and might warrant adding there too